addons/mrp/models/stock_rule.py

```diff
@@ -30,28 +30,31 @@ def _onchange_action_operation(self):
         return {'domain': domain}
 
     @api.multi
-    def _run_manufacture(self, product_id, product_qty, product_uom, location_id, name, origin, values):
-        Production = self.env['mrp.production']
-        ProductionSudo = Production.sudo().with_context(force_company=values['company_id'].id)
-        bom = self._get_matching_bom(product_id, values)
-        if not bom:
-            msg = _('There is no Bill of Material found for the product %s. Please define a Bill of Material for this product.') % (product_id.display_name,)
-            raise UserError(msg)
-
-        # create the MO as SUPERUSER because the current user may not have the rights to do it (mto product launched by a sale for example)
-        production = ProductionSudo.create(self._prepare_mo_vals(product_id, product_qty, product_uom, location_id, name, origin, values, bom))
-        production.move_raw_ids = self.env['stock.move'].create(production._get_moves_raw_values())
-        production.action_confirm()
-        origin_production = values.get('move_dest_ids') and values['move_dest_ids'][0].raw_material_production_id or False
-        orderpoint = values.get('orderpoint_id')
-        if orderpoint:
-            production.message_post_with_view('mail.message_origin_link',
-                                              values={'self': production, 'origin': orderpoint},
-                                              subtype_id=self.env.ref('mail.mt_note').id)
-        if origin_production:
-            production.message_post_with_view('mail.message_origin_link',
-                                              values={'self': production, 'origin': origin_production},
-                                              subtype_id=self.env.ref('mail.mt_note').id)
+    def _run_manufacture(self, procurements_list):
+        productions_values = []
+        for procurement, rule in procurements_list:
+            bom = self._get_matching_bom(procurement.product_id, procurement.values)
+            if not bom:
+                msg = _('There is no Bill of Material found for the product %s. Please define a Bill of Material for this product.') % (procurement.product_id.display_name,)
+                raise UserError(msg)
+
+            # create the MO as SUPERUSER because the current user may not have the rights to do it (mto product launched by a sale for example)
+            productions_values.append(rule._prepare_mo_vals(*procurement, bom))
+        production_ids = self.env['mrp.production'].create(productions_values)
+        self.env['stock.move'].create(production_ids._get_moves_raw_values())
+
+        for production in production_ids:
+            production.action_confirm()
```

Comment:
you could run action_confirm directly for all production i think
won't matter much at the moment but you could also improve action_confirm of the mo